src/components/AssetRegistration/AssetRegistration.tsx

```diff
@@ -0,0 +1,197 @@
+    );
+
+    // check if the asset is LSP7 or LSP8
+    const checkStandard = async (asset: string) => {
+      const isLSP7 = await supportsInterface(
+        asset,
+        INTERFACE_IDS.LSP7DigitalAsset
+      );
+      const isLSP8 = await supportsInterface(
+        asset,
+        INTERFACE_IDS.LSP8IdentifiableDigitalAsset
+      );
+      return isLSP7
+        ? INTERFACE_IDS.LSP7DigitalAsset
+        : isLSP8
+          ? INTERFACE_IDS.LSP8IdentifiableDigitalAsset
+          : false;
+    };
+
+    // encode transaction data
+    const issuedAssetMap: {
+      keyName: string;
+      dynamicKeyParts: string;
+      value: string[];
+    }[] = await Promise.all(
+      allAssets.map(async (asset: string, index: number) => {
+        const assetInterface = await checkStandard(asset);
+
+        return {
+          keyName: 'LSP12IssuedAssetsMap:<address>',
+          dynamicKeyParts: asset,
+          value: [assetInterface, ERC725.encodeValueType('uint128', index)],
+        };
+      }, [])
+    );
+
+    const { keys: lsp12DataKeys, values: lsp12Values } = erc725.encodeData([
+      {
+        keyName: 'LSP12IssuedAssets[]',
+        value: allAssets,
+      },
+      ...issuedAssetMap,
+    ]);
+
+    const signer = await provider?.getSigner();
+
+    // Universal Profile Contract Instance
+    const myUniversalProfileContract = new ethers.Contract(
+      account,
+      UniversalProfileContract.abi,
+      signer
+    );
+
+    setTransaction({ ...transaction, processing: true });
+
+    myUniversalProfileContract
+      .setDataBatch(lsp12DataKeys, lsp12Values)
+      .then((tx: any) => {
+        setTransaction({ processing: false, receipt: tx.hash });
+      })
+      .catch((error: any) => {
+        setTransaction({ ...transaction, processing: false });
+        console.error(error);
+      });
+  };
+
+  const { processing, receipt } = transaction;
+
+  return (
+    <div className="max-w-[432px] flex flex-col justify-center items-center">
+      <div className="flex flex-col">
+        {allAssets.map((asset) => (
+          <div key={asset}>{asset}</div>
+        ))}
+      </div>
+      <form onSubmit={(e) => e.preventDefault()}>
+        <input
+          className="w-[420px] bg-gray-200 text-gray-700 font-bold my-2 p-2 rounded"
+          type="text"
+          id="asset"
+          name="asset"
+          placeholder="Asset Address"
+          value={newAsset}
+          onChange={(e) => setNewAsset(e.target.value)}
+        />
+        <div className="flex justify-center items-center">
+          <button
+            className={
+              !allAssets.length
+                ? 'm-2 bg-gray-300 text-gray-400 font-bold py-2 px-4 rounded cursor-not-allowed'
+                : 'm-2 bg-lukso-pink text-white font-bold py-2 px-4 rounded'
+            }
+            onClick={() => {
+              registerAssets();
+            }}
+          >
+            Set LSP12IssuedAsset
```

Comment:
As above, standard name is in the plural.

```suggestion
            Set LSP12IssuedAssets
```